### DISCUSSION TITLE
How can we determine if an entity instance is managed by unitOfWork？

### DISCUSSION THREAD

**Discussion by Lydanne** [mikro-orm/mikro-orm#2334] [Q&A] [answered]
I want to implement a save method that I will persistAndFlush when not managed if managed I flush directly
```ts

export abstract class CommonEntityRepository<
  TEntity extends CommonEntity,
> extends EntityRepository<TEntity> {

  async save(entity: TEntity): Promise<TEntity> {
    if (entity.id) {
      await this.flush();
    } else {
      await this.persistAndFlush(entity);
    }
    return entity;
  }


```

**Comment by B4nan** [comment, marked as answer]
1. you can persist managed entity too, it is a no-op, so in your example you could just go with `persistAndFlush`.
2. you can tell this based on existence of `wrap(entity, true).__originalEntityData` (internal API, so it can break over time)

But it does not make much sense, you should aim to flush once per handler, not once per entity. Each flush is a separate transaction, having one transaction per one insert/update call kinda breaks the whole idea.

**Reply by Lydanne** [reply]
I don't want to use mongo transactions, can I implement save more elegantly

**Reply by B4nan** [reply]
Oh right, in mongo it is a different story :]

**Reply by Lydanne** [reply]
what，What am I supposed to do about it 😭

**Reply by B4nan** [reply]
Nothing, I am just saying that it's kinda fine to do this in mongo.

**Reply by Lydanne** [reply]
ok ok，Thank you 👍🏻
